datapower-docker-compose-dp-mq/datapower-docker-compose-dp-mq.mkd

```diff
@@ -0,0 +1,168 @@
+src_curldriver_1 exited with code 0
+
+```
+`src_soapui_1` has assertions set up to validate the message being returned. When the assertions are met, a `200 - OK` message is returned to `curldriver_1` as can be seen in the output above.
+
+Depending on the system on which you are deploying, the first message or two may fail to be processed as the environment may still be building. This is easily detectable in the console logs.
+
+After the first successful response is received the curldriver script exits. You may end the test with the docker command ctrl+c, which will exit the application and stop the containers. To rerun the test, run the `docker-compose up` command again.
+
+## Request Flow- Meet "bob"
+Here I'll outline the travels of the request message, `bob`, through the message flow created in the docker-compose environment. 
+
+First from the Dockerfile, the `src_curldiriver` image is configured to run the following shell script which sends a curl command via HTTP to `src_soapui`.
+```
+#!/bin/bash
+# set env vars to modify operation
+# TIMEOUT=<time in seconds> for amount of time to continue to retry
+# CONTINUOUS=true to continuously test, otherwise exits immediately on success
+
+if [ "$TIMEOUT" -a "$TIMEOUT" -gt 0 ]
+then
+  STOPTIME=$(( $(date +%s) + $TIMEOUT ))
+else
+  STOPTIME=9999999999
+fi
+
+RC=1
+
+while [ $(date +%s) -lt $STOPTIME ]
+do
+  if curl --silent --insecure --form "project=@/usr/local/bin/grow-soapui-project.xml" --form "suite=sendRequestGrow" http://soapui:3000 | grep "200 OK"
+  then
+    RC=0
+    if [ "$CONTINUOUS" != "true" ]
+    then
+      echo "SUCCESS! Exiting"
+      exit $RC
+    fi
+  else
+    echo curl error $?
+  fi
+  sleep 5
+done
+```
+The script curls a SoapUI project named `grow-soapui-project.xml`, located in the /usr/local/bin/ directory, to `src_soapui_1`. The script is set to send a request message every five seconds until a successful transaction is completed. Note that the console log may show a few request failures as the application comes up.
+
+`src_soapui_1` accepts the curl request and processes the test defined in the SoapUI project suite. This is where `bob` is created as an XML message and sent to `src_datapower_1`.
+
+`src_datapower_1` accepts `bob` from `src_soapui_1`. The processing policy configured in the multi-protocol gateway, in the application domain, retrieves various stylesheets and schema from `src_http_1` and transforms `bob`. The policy itself is simple. As `bob` passes through each rule in the policy, his number of nodes and elements increase. At the end, `bob` is a message that reports what size he was in bytes at the end of the trip through DataPower. Bob is then routed to the `src_mq_1` back end.
+
+`bob` is placed on a queue in `src_mq_1` and his travel could very well end here; however, to avoid the queue getting full on a long running test, I set up a dummy service on `src_datapower_1` that clears the queue and throws the message away. 
```

Comment:
After the transformed message is placed on a queue in `src_mq_1` that could have been the end of the story. However, to avoid the queue getting full on a long running test, I set up a dummy service on `src_datapower_1` that clears the queue and throws the message away.